### DISCUSSION TITLE
T-Pot 22.04.0

### DISCUSSION THREAD

**Discussion by t3chn0m4g3** [telekom-security/tpotce#1068] [Announcements]
# Release Notes / Changelog
T-Pot 22.04.0 is probably the most feature rich release ever provided with long awaited (wanted!) features readily available after installation. 

## New Features
* **Distributed** Installation with **HIVE** and **HIVE_SENSOR**
* **ARM64** support for all provided Docker images
* **GeoIP Attack Map** visualizing Live Attacks on a dedicated webpage
* **Kibana Live Attack Map** visualizing Live Attacks from different **HIVE_SENSORS**
* **Blackhole** is a script trying to avoid mass scanner detection 
* **Elasticvue** a web front end for browsing and interacting with an Elastic Search cluster
* **Ddospot** a honeypot for tracking and monitoring UDP-based Distributed Denial of Service (DDoS) attacks
* **Endlessh** is a SSH tarpit that very slowly sends an endless, random SSH banner
* **HellPot** is an endless honeypot based on Heffalump that sends unruly HTTP bots to hell
* **qHoneypots** 25 honeypots in a single container for monitoring network traffic, bots activities, and username \ password credentials
* **Redishoneypot** is a honeypot mimicking some of the Redis' functions
* **SentryPeer** a dedicated SIP honeypot
* **Index Lifecycle Management** for Elasticseach indices is now being used

## Upgrades
* **Debian 11.x** is now being used for the T-Pot ISO images and required for post installs
* **Elastic Stack 8.x** is now provided as Docker images

## Updates
* **Honeypots** and **tools** were updated to their latest masters and releases
* Updates will be provided continuously through Docker Images updates 

## Breaking Changes
* For security reasons all Py2.x honeypots with the need of PyPi packages have been removed: **HoneyPy**, **HoneySAP** and **RDPY**
* If you are upgrading from a previous version of T-Pot (20.06.x) you need to import the new Kibana objects or some of the functionality will be broken or will be unavailabe
* **Cyberchef** is now part of the Nginx Docker image, no longer as individual image
* **ElasticSearch Head** is superseded by **Elasticvue** and part the Nginx Docker image
* **Heimdall** is no longer supported and superseded with a new Bento based landing page
* **Elasticsearch Curator** is no longer supprted and superseded with **Index Lifecycle Policies** available through Kibana.

# Thanks & Credits
* @ghenry, for some fun late night debugging and of course SentryPeer!
* @giga-a, for adding much appreciated features (i.e. JSON logging, 
X-Forwarded-For, etc.) and of course qHoneypots! 
* @sp3t3rs, @trixam, for their backend and ews support!
* @tadashi-oya, for spotting some errors and propose fixes!
* @tmariuss, @shaderecker for their cloud contributions!
* @vorband, for much appreciated and helpful insights regarding the GeoIP Attack Map!
* @yunginnanet, on not giving up on squashing a bug and of course Hellpot!

... and many others from the T-Pot community by opening valued issues and discussions, suggesting ideas and thus helping to improve T-Pot!

<hr /><em>This discussion was created from the release <a href='https://github.com/telekom-security/tpotce/releases/tag/22.04.0'>T-Pot 22.04.0</a>.</em>

**Comment by yunginnanet** [comment]
Wow. I didn't even notice that you were the top contributor to this project. This project taught me a lot about honeynets. It's truly an honor.

**Reply by t3chn0m4g3** [reply]
Thank you!

**Comment by ghenry** [comment]
An absolute pleasure working with you @t3chn0m4g3 Peer to Peer is live now in the [v1.4.0 container](https://hub.docker.com/layers/sentrypeer/sentrypeer/sentrypeer/v1.4.0/images/sha256-0df300cec8c8e13a0a01b15f61a3683f9451eb546726f701714a4358f853d462?context=explore) too ([my Dockerfile updated](https://github.com/SentryPeer/SentryPeer/blob/main/Dockerfile)) as I packaged up [OpenDHT for Alpine](https://github.com/SentryPeer/OpenDHT-packaging/tree/main/alpinelinux). Have a play!

Thanks!

**Reply by t3chn0m4g3** [reply]
Thank you! As always, likewise!
Just added to the [shortlist](https://github.com/telekom-security/tpotce/projects/5#column-15061838) 😅

**Comment by lcia-projects** [comment]
very interested in the HIVE + HIVE_SENSOR setup.. is there any more information on that? searched discussion and issues.. nothing..

**Reply by t3chn0m4g3** [reply]
It is documented in the [Readme](https://github.com/telekom-security/tpotce#readme).

**Comment by mibmit** [comment]
Deployed a couple of new T-Pot's  - Great - seem to be working well, and nice visualisations.

I am getting 

Raise network interfaces : Failed to start 

All seems to be working fine, collecting logs and binaries.

**Reply by t3chn0m4g3** [reply]
It is documented in the [Readme](https://github.com/telekom-security/tpotce#readme) under Known Issues.

**Comment by ehnwebmaster** [comment]
Fantastic, awesome work!

 Now my T-Pot is working fine (without emulation) on a RaspBerry Pi 4! Thank you for adding ARM64 support to all docker images.

**Reply by t3chn0m4g3** [reply]
Thank you for the great feedback. Would you mind sharing some details how you made it work?

**Reply by ehnwebmaster** [reply]
Frist I was thinking using tpot_arm64.iso (debian net install) but finally used  Raspberry Pi OS (Kernel version: 5.15 based on Debian version: 11 bullseye) and do the post-installation primarlly because want to set led offs using the overlays and booting from USB pendrive instead of using SD card

Post install works fine:
```
git clone https://github.com/telekom-security/tpotce
cd tpotce/iso/installer/
./install.sh --type=user
```

Everything works flawless so all packages from Debian 11 arm64 are  available on Raspi OS aarch64. No errors, no extra config.

Even I can use "standard" installation using Kibana, logstash, elasticsearch) (RaspBerry Pi 4 8GB, full ram used) but right now I'm using just the "sensor" with just 2GB ram in use). Just noticed sometimes high cpu usage from ewsposter (just normal, because it's reporting many IP's)

`MiB Mem :   7811.5 total,   2173.0 free,   2312.4 used,   3326.1 buff/cache`

Used and Left space:

```
Filesystem      Size  Used Avail Use% Mounted on
/dev/root       114G   15G   95G  14% /

```

Last time (using previous version of Tpot without arm support) installing and configuring Tpot wasn't easy...... I must use docker emulator based on qmeu to emulate all docker containers using:
 
`docker run --rm --privileged aptman/qus -s -- -p x86_64`

Then everything works (minus dioanaea last version reporting SMB failed due to "setgroups dropping groups failed"), but there was high cpu usage due to emulation (quite normal)

Log install.err

> info: Trying to set 'docker.io/restart' [boolean] to 'true'
> info: Loading answer for 'docker.io/restart'
> info: Trying to set 'debconf/frontend' [select] to 'noninteractive'
> info: Loading answer for 'debconf/frontend'
> W: --force-yes is deprecated, use one of the options starting with --allow instead.
> W: --force-yes is deprecated, use one of the options starting with --allow instead.
> npm WARN deprecated har-validator@5.1.5: this library is no longer supported
> npm WARN deprecated s3signed@0.1.0: This module is no longer maintained. It is provided as is.
> npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParam>
> npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certai>
> npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
> Cloning into '/opt/tpot'...
> Created symlink /etc/systemd/system/multi-user.target.wants/tpot.service → /etc/systemd/system/tpot.service.
> sed: can't read /etc/default/grub: No such file or directory
> sed: can't read /etc/default/grub: No such file or directory
> ./install.sh: line 865: update-grub: command not found

**Reply by t3chn0m4g3** [reply]
Thank you for sharing!

Will be ordering RasPi 4+ with 8GB and check out if support can be even improved further.

**Reply by dgarcia88** [reply]
I tried to install on pi4 with 8GB ram
Issues i have found:
In overview "dhcpd service keep failing"
Kitana won't open other link's works fine (maybe i did something wrong?)
CPU usage always around 80~100%

I gonna reinstall all over again and report back :)

**Reply by dgarcia88** [reply]
All seems running fine on Raspberry Pi it takes arround ~40min to all be working.

**Comment by damakusan** [comment]
Hello! 
I have an error in installation progress. Can you help me?

![Failed install](https://user-images.githubusercontent.com/110384558/182678808-9b32efe0-4524-416e-a111-68a81402f4c3.png)

**Reply by yunginnanet** [reply]
I would recommend using the issues section for this sort of thing: https://github.com/telekom-security/tpotce/issues

**Reply by damakusan** [reply]
Ok, thank you

**Comment by ghenry** [comment]
Afternoon. Not sure if anyone has reported, but the tpot_amd64.iso is actually a 32bit arch :-)

**Reply by ghenry** [reply]
I think it's the unattended conf doing this. If I run it as 32bit, the kernel is 64bit, so I need to change to manual on VirtualBox

**Reply by t3chn0m4g3** [reply]
The preseed has no influence on the architecture. The ISO is [built](https://github.com/telekom-security/tpotce/blob/2641d1e74306fafa9c43f08fb6927e3d9781c936/makeiso.sh#L75-L83) against the Debian Netiso with AMD64 or ARM64.

**Reply by ghenry** [reply]
Must be how VirtualBox is doing things as it makes the option readonly unless you choose manual (via the GUI anyway).

Thanks.

**Comment by felipemor** [comment]
Hello ! I install my tpot in Vbox but now i dont see kibana app, do you have a solution for this ? Hugs

**Reply by t3chn0m4g3** [reply]
Kibana will only be shown if the service has been started already. If everything else works fine a reload of the landing page should show the Kibana link.

**Reply by felipemor** [reply]
Hello @t3chn0m4g3, i hope you're fine.

Thanks a lot for the support, i execute the instructions and now i have sucessfully on service.

Regards.

**Comment by TPOTinstallatiobnir** [comment]
Hi
I Wanna to install topot on ESXi (HP Server) which version should be install ?

**Comment by Cristobalvaldivia** [comment]
Hello!
I have an error in installation progress. Can you help me?
<img width="368" alt="error" src="https://github.com/telekom-security/tpotce/assets/103283012/d1d44b90-0831-4fcd-989d-400a2096b6ad">

**Reply by t3chn0m4g3** [reply]
I just updated the [release](https://github.com/telekom-security/tpotce/releases/tag/22.04.0).

**Comment by Cristobalvaldivia** [comment]
I performed the installation but the kibana option does not appear. 

<img width="959" alt="kibana" src="https://github.com/telekom-security/tpotce/assets/103283012/a0d39a10-0b08-4dfa-bcb6-77b24d645183">

**Reply by t3chn0m4g3** [reply]
It only appears once Kibana has been started successfully. Reload?

**Comment by jmargon** [comment]
I performed the installation in a VM, but the nginx container don't start.
![image](https://github.com/telekom-security/tpotce/assets/153286367/765093a9-c3fd-43cf-820e-1e3c503fd51f)
What's going on? It don't start, it's always restarting.
![image](https://github.com/telekom-security/tpotce/assets/153286367/8df1598d-04d7-49b3-bff8-0c90a853d9d4)

**Comment by kssjoe** [comment]
It's it worth installing Defender on the VM?

**Comment by Cristobalvaldivia** [comment]
I have the following problem when configuring the t-pot with the manual network i get the following error what can i do and i am installing it in Vmware esxi 
<img width="360" alt="Capturadddd" src="https://github.com/telekom-security/tpotce/assets/103283012/f72d6bf7-f5b3-4525-b9b7-588e7b7ccbbb">

**Comment by famirapt** [comment]
this project is actually inspiring, which am currently using as for my final year project pls can you explain how i can download and install it on virtual machine

**Comment by famirapt** [comment]
Yes, I have import It on VMware, but need your assistance in complete the
configuration and analyzing attacks,
My project is on stimulating attack and detecting response from the
honeypot , if you could help me in recommending finished project documents
why I could find how to stimulate attack on my honey and analyze it.

**Comment by morenblakk** [comment]
after installing tpot, i cannot login. all username(tsec) and password is not working. including the username that i set while installing. i  have uninstalled and reinstalled it more than 6 times now, each time it comes up with different names like briefrank login, usualuser login, etc. i need help folks
